python/ray/serve/_private/replica.py

```diff
@@ -713,6 +718,26 @@ async def initialize(self, deployment_config: DeploymentConfig):
         except Exception:
             raise RuntimeError(traceback.format_exc()) from None
 
+    async def update_replica_rank(self, rank: int, world_size: int) -> bool:
+        """Update replica rank with the provided rank value.
+
+        This is called by the deployment state to notify replicas of their
+        current rank after autoscaling operations.
+
+        Args:
+            rank: The current rank of this replica.
+            world_size: The current number of running replicas in the deployment.
+
+        Returns:
+            True if rank update was successful, False otherwise.
+        """
+        self._rank = rank
+        self._world_size = world_size
```

Comment:
why is `_rank` and `_world_size` attributes of the object instead of set directly through `self._set_internal_replica_context`